src/main/java/io/stargate/sgv2/jsonapi/api/security/ErrorChallengeSender.java

```diff
@@ -59,8 +85,13 @@ public Uni<Boolean> apply(RoutingContext context, ChallengeData challengeData) {
           .headers()
           .set(HttpHeaders.CONTENT_LENGTH, String.valueOf(response.getBytes().length));
 
-      // always set status to 200
-      context.response().setStatusCode(200);
+      // Return the status code from the challenge data
+      context.response().setStatusCode(challengeData.status);
+
+      // Add metrics
+      Tags tags = Tags.of(apiTag, commandTag, statusCodeTag, errorTag);
+      // record
+      meterRegistry.counter("http_server_requests_custom_seconds_count", tags).increment();
```

Comment:
Not sure if there's a good place to define these constants? If not, inlining is fine.

Reply:
Will add it to the Constants